### ISSUE TITLE
Could openids and accounts change hands? 

### ISSUE THREAD

**Issue by garemoko** [adlnet/xAPI-Spec#703]
This might be a 2.0 issue. 

We say in relation to the mbox IFI that ```Only email addresses that have only ever been and will ever be assigned to this Agent, but no others, SHOULD be used for this property and mbox_sha1sum.```

Is there any reason why this same principal (as a SHOULD) doesn't apply to the other IFIs?

**Comment by canweriotnow** [comment]
Why yes they could. Which is why I've been a voice in the wilderness agitating for an identifier equivalency standard since I saw this spec the very first time. It could be an `actor` `IS` `actor` equivalency statement, saying foo@bar.com is foo on twitter and foo on github and openid foo, and then a series of voiding statements retracting those... Of course, this assumes an LRS that can deal with a state in time. 

Lacking that, I guess you're SOL.

But for the love of all that is holy...

THINGS CHANGE. WE NEED TO COPE WITH THIS UNFORTUNATE AND TRAGIC FACT OF HUMAN EXISTENCE.

**Comment by canweriotnow** [comment]
How about this, @garemoko:

An LRS **MUST** be able to reproduce its current state as well as recreate its state at any given point in the past, given the insistence on the **immutability** of xAPI Statements, yet the intransigent mutability of the world at large.

Let's include that in 1.0.3, shall we?

I mean, we designed our LRS that way, didn't everybody?

**Comment by canweriotnow** [comment]
So, really, a well-designed LRS shouldn't have a problem with changes in... anything, right? 

And then assigning equality between identifiers, at least for a period of time, shouldn't be a problem? So we can co ahead with that?

HALLELUJAH!!!

**Comment by garemoko** [comment]
You're not the only voice, I think it just comes down to priorities. From memory, I believe @fugu13 first proposed something like an 'is' verb a while back and I actually recently poked @andyjohnson for an ADL verb for this very recently for a specific adopter. The adopter's use case turned out to be slightly different in the end and they went down another route. 

I'm not sure I follow that MUST you suggested. Remember that statements are about things that have happened. The past isn't mutable (we can maybe disagree on that philosophically, but it's true enough for practical purposes!)

**Comment by canweriotnow** [comment]
@garemoko Thanks, I'll have to reach back out to @fugu13 and see where he is on that, I know it's come up before.

(I know I'm not the only voice, I just like the literary reference and it feels like that sometimes).

I think it's equally a distinction of philosophy and physics on whether the past _itself_ is mutable, but records of the past within a system like an LRS certainly are (well, let's say they are by default, personally working with a system designed to avoid this potentiality), so if I wanted to assert that Hitler died in 1962 in a SOR responsible for the past century of human history, and that was a trusted system, then for all intents and purposes, if I could pull off that very clever hack, Hitler did, indeed, die in 1962.

Putting aside the mutability of the past, our perception of it and, _a fortiori,_ the historical record, are mutable _par excellence._

**Comment by canweriotnow** [comment]
And just for @aaronesilvers (and @garemoko and @fugu13 and @andyjohnson and everyone else)...

![](http://38.media.tumblr.com/fd1ddb441c88da0eba2003250a4e00e9/tumblr_n8h5qk48nF1tesnd4o1_250.gif)

**Comment by garemoko** [comment]
I wondered if somebody would bring up killing Hitler. 

I think that the changing history use case should be rare enough that either voiding or alternative more trusted authorities should be sufficient.

In The Doctor's case, I'd expect that the TARDIS should handle editing of LRS records in the same way it handles changing of actual historical events without the need for interoperability specifications. That said, perhaps some kind of standardization would be helpful for the avoidance of paradox. Either way, it's out of scope.

**Comment by canweriotnow** [comment]
@garemoko I don't think this is sufficiently reliable; recall that in [The Deadly Assassin](https://en.wikipedia.org/wiki/The_Deadly_Assassin), the Master had managed to remove every record of himself not only from the TARDIS databanks, but from The Matrix itself, the central repository of all Time Lord knowledge.

Recall also that when the Doctor suggested this might be the case, that The Matrix might have been "hacked," Engin assures the Doctor that Gallifrey had abandoned such electromechanical technology as we employ when the universe was less than half its present age. 

Therefore, one might extrapolate that it will be 2n (where n is the current age of the universe) when we can expect the mutability of the past and the security of record of any changes to the past to become an "edge case", unless we gain access to Gallifreyan technology in some currently unlikely timeline, as Gallifrey has been sealed off in a bottle universe.

**Comment by blakeplock** [comment]
Are statements "about things that have happened" or are they artifacts of a certain type of observation that has happened? The record of the observation of historical events is obvious mutable. But it doesn't have to be.

**Comment by davidpesce** [comment]
Hands down, the best issue thread I've read.

**Comment by canweriotnow** [comment]
@blakeplock I think by definition they are artifacts, representing a type of machine-facilitated observation. And as you point out, the record can always be changed, unless there is a mechanism in place preventing it from happening. 

For instance, if an LRS were backed by a SQL DB or even something like Mongo, it would be simple enough to use direct DB access to mutate the state of stored records circumventing whatever application-level safeguards were in place to impose immutability or even implement an audit trail. 

Reliable immutability requires different persistence technologies, which is what I was sort of implying in the facetious MUST I proposed to @garemoko.

But that's somewhat tangential to this particular issue, although essential (IMHO) to the viability of xAPI.
